numbert/reranker.py

```diff
@@ -0,0 +1,237 @@
+'''
+Module for providing python interface to numBERT re-rankers
+'''
+
+import os
+os.environ["JAVA_HOME"] = "/usr/lib/jvm/java-11-openjdk-amd64"
+
+from pyserini.search import pysearch
+
+from .utils_reranker import load_model, online_eval, Hit
+from .scripts.utils_numbert import processors, output_modes, convert_examples_to_features
+
+class Reranker(object):
+    '''
+    Base class for Passage Ranking Models
+    (Defaults to the behavior of Anserini's search)
+    Parameters
+    ----------
+    index_dir : str
+        Path to Lucene index directory
+    '''
+    def __init__(self, index_dir):
+        self.simple_searcher = pysearch.SimpleSearcher(index_dir)
+
+    def rerank(self, q, k=100, t=-1, threads=1):
+        '''
+        Parameters
+        ----------
+        q : str
+            Query string
+        k : int
+            Number of hits to return from pyserini's SimpleSearcher
+        t : int
```

Comment:
Maybe add a description for `threads`?

Reply:
Fixed in recent commit! Thanks for pointing it out, was missing the thread arguments in quite a few places!